Doc edits for CAF (#36)
aws-quickstart/quickstart-f5-big-ip-virtual-edition: docs/deployment_guide/partner_editable/pre_deployment.adoc

This Quick Start requires:

> * The appropriate permission in AWS to launch AWS CloudFormation templates. You must be an IAM user with the Administrator Access policy attached and have permission to create the objects contained in this Quick Start, including VPCs, Elastic IP addresses, EC2 instances, IAM roles and Instance Profiles, SSE-S3 encrypted S3 buckets, SSH Key Pairs, AWS Secrets Manager secrets, etc.

Reviewer: ```suggestion
* Appropriate AWS permissions to launch AWS CloudFormation templates. You must be an IAM user with administrator access and have permission to create the objects contained in this Quick Start. Objects include but are not limited to VPCs, elastic IP addresses, EC2 instances, IAM roles and instance profiles, SSE–S3 encrypted S3 buckets, SSH key pairs, and AWS Secrets Manager secrets.
```

> * This solution requires an SSH key pair in AWS for management access to {partner-product-short-name}. For more information about creating and/or importing the key pair in AWS, refer to https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html[Amazon EC2 key pairs and Linux instances^]. If one is not specified, one will be created for you named `BigIpSshKeyPair`.

Reviewer: ```suggestion
* This deployment requires an SSH key pair in AWS for management access to {partner-product-short-name}. For more information about creating and importing the key pair in AWS, refer to https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html[Amazon EC2 key pairs and Linux instances^]. If a key pair is not specified, `BigIpSshKeyPair` is created for you.
```

> * This solution requires AWS secret stored in https://aws.amazon.com/secrets-manager/[AWS Secrets Manager^] containing the password used to access and cluster the HA pair. If one is not specified, one will be created for you named `BigIpSecret` with an auto-generated password. To obtain the secret value, you can run the following https://docs.aws.amazon.com/cli/latest/reference/secretsmanager/get-secret-value.html[AWS cli command^], changing the region and secret name as needed:

Reviewer: ```suggestion
* This deployment requires an AWS secret stored in https://aws.amazon.com/secrets-manager/[AWS Secrets Manager^] that contains the password to access the cluster. If you don't specify a secret, one is created for you named `BigIpSecret` with an auto-generated password. To obtain the secret value, run the following https://docs.aws.amazon.com/cli/latest/reference/secretsmanager/get-secret-value.html[CLI command^], but change the Region and secret name as needed:
```

    aws secretsmanager get-secret-value --secret-id BigIpSecret --query "SecretString" --output text --region us-east-1

> *** For more information, refer to the https://aws.amazon.com/secrets-manager/[AWS CLI secretsmanager documentation^].

Reviewer: ```suggestion
*** For more information, refer to the https://aws.amazon.com/secrets-manager/[AWS Secrets Manager^].
```

> *** To update the password to your organization's policy or requirements, see:

Reviewer: ```suggestion
*** To update your organization's password, refer to the following resources:
```

**** https://support.f5.com/csp/article/K13121[K13121: Changing system maintenance account passwords^]

> **** https://support.f5.com/csp/article/K15497[K15497: Configuring a secure password policy for the BIG-IP system^]

Reviewer: ```suggestion
**** https://support.f5.com/csp/article/K15497[K15497: Configuring a secure password policy for the BIG-IP system (11.x - 16.x)^]
```

==== Configuration notes ====

* The Quick Start template provides an initial deployment for an infrastructure use case, which means that it doesn't support managing the entire deployment exclusively via the template's update-stack feature.
+

> This Quick Start uses the `cloud-init` package to send the instance user data, which is only used to provide an initial {partner-product-short-name} configuration and not as the primary configuration API for a long-running platform. You can use the update-stack feature to update certain cloud resources, however the {partner-product-short-name} configuration needs to align with cloud resources, such as IPs and NICs. Therefore, updating one resource without the other can result in inconsistent states, while updating other resources, like the *bigIpImageId* or *bigIpInstanceType* parameters, can trigger an entire instance re-deloyment.

Reviewer: ```suggestion
This Quick Start uses the `cloud-init` package to send instance user data, which is only used to provide an initial {partner-product-short-name} configuration but not the primary configuration API for a long-running platform. You can use the update-stack feature to update certain cloud resources, but the {partner-product-short-name} configuration must align with cloud resources, such as IPs and NICs. Therefore, updating one resource without the other can result in inconsistent states, such as when updating the `bigIpImageId` and `bigIpInstanceType` parameters, which can trigger an instance redeployment.
```

+

> For example, to upgrade software versions, traditional in-place upgrades and configuration management are required. For details, refer to the link:#_support[Support] section.

Reviewer: ```suggestion
For example, to upgrade software versions, traditional in-place upgrades and configuration management are required. For more information, refer to the link:#_support[Support] section.
```

> * This solution requires the BIG-IPs to have Internet access to download the required software packages. If you are deploying this solution in a private subnet, you will need to configure a NAT gateway or NAT instance to allow the BIG-IPs to access the internet. See the

Reviewer: ```suggestion
* This deployment requires internet access to download the required software packages. If you deploy this Quick Start to a private subnet, you must configure a NAT gateway or NAT instance to allow the BIG-IPs to access the internet. For more information, refer to the
```

> link:#_security[Security] section for additional information.

Reviewer: ```suggestion
link:#_security[Security] section.
```

> * This solution uses https://clouddocs.f5.com/products/extensions/f5-cloud-failover/latest/[Cloud Failover Extension (CFE)^] to provide failover functionality. By default, this solution creates an S3 bucket for use by the F5 Cloud Failover Extension (CFE). The bucket is configured with AWS SSE-S3 encryption enabled and the IAM role is set to enforce AES256 encryption. See https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingServerSideEncryption.html[Protecting data using server-side encryption with Amazon S3-managed encryption keys (SSE-S3)^] for more information. See Cloud Failover Extension https://clouddocs.f5.com/products/extensions/f5-cloud-failover/latest/userguide/aws.html#aws-define-storage[documentation^] for more information.

Reviewer: ```suggestion
* To provide failover functionality, this Quick Start uses the https://clouddocs.f5.com/products/extensions/f5-cloud-failover/latest/[Cloud Failover Extension (CFE)^]. By default, this deployment creates an S3 bucket for use by the F5 CFE. The bucket is configured with AWS SSE–S3 encryption enabled, and the IAM role is set to enforce AES-256 encryption. For more information, refer to https://docs.aws.amazon.com/AmazonS3/latest/userguide/UsingServerSideEncryption.html[Using server-side encryption with Amazon S3-managed encryption keys (SSE-S3)^] and https://clouddocs.f5.com/products/extensions/f5-cloud-failover/latest/userguide/aws.html#aws-define-storage[AWS CFE Prerequisites^].
```

> * This template can send non-identifiable statistical information to F5 Networks to help us improve our templates. You can disable this functionality for this deployment only by supplying **false** for the value of the **allowUsageAnalytics** input parameter, or you can disable it system-wide by setting the **autoPhonehome** system class property value to false in the F5 Declarative Onboarding declaration. See the link:#_customize_the_big_ip_ve_configuration[Customizing the BIG-IP Configuration] section below and link:#_security[Security] section for additional information.

Reviewer: ```suggestion
* This template sends nonidentifiable statistical information to F5 Networks. You can disable this functionality by setting the `allowUsageAnalytics` parameter to **false**, or you can disable it system wide by setting the `autoPhonehome` system class property value to **false** in the F5 onboarding declaration. For more information, refer to the link:#_customize_the_big_ip_ve_configuration[Customize the BIG-IP VE configuration] and link:#_security[Security] sections.
```

=== Customize the {partner-product-short-name} configuration

> If needed, customize the {partner-product-short-name} configuration. The initial {partner-product-short-name} configurations are passed through the https://github.com/f5networks/f5-bigip-runtime-init[F5 BIG-IP Runtime Init^] configuration files. These files are the primary mechanisms to customize the initial {partner-product-short-name} configurations for your deployment. They also allow for large amounts of {partner-product-short-name} customizations. Customizing the {partner-product-short-name} configurations generally involves updating and rehosting these files and passing their location through the *bigIpRuntimeInitConfig* template parameters.

Reviewer: ```suggestion
The initial {partner-product-short-name} configurations pass through the https://github.com/f5networks/f5-bigip-runtime-init[F5 BIG-IP Runtime Init^] configuration files. These files are the primary mechanism to customize the initial {partner-product-short-name} deployment configuration. Customizing the {partner-product-short-name} configurations generally involves updating and rehosting these files and passing their location through the `bigIpRuntimeInitConfig` template parameters.
```

> By default, the Quick Start deploys a minimal 2NIC PAYG configuration into the default Quick Start network topology using the following configuration files in this repo's https://github.com/aws-quickstart/quickstart-f5-big-ip-virtual-edition-ha/tree/main/declarations[/declarations^] folder:

Reviewer: ```suggestion
By default, this Quick Start deploys a minimal `2NIC PAYG` configuration to the default Quick Start network topology using the following configuration files from the https://github.com/aws-quickstart/quickstart-f5-big-ip-virtual-edition-ha/tree/main/declarations[declarations^] folder:
```

** `runtime-init-conf-2nic-payg-instance01.yaml`
** `runtime-init-conf-2nic-payg-instance02.yaml`

> A minimal baseline high availability (HA) failover cluster is deployed. No virtual services are provisioned.

Reviewer: ```suggestion
A baseline high-availability failover cluster is deployed. No virtual services are provisioned.
```

> Optionally, you can deploy an example web application firewall (WAF)-protected web service to illustrate how services work. The example application deploys web application instances and a WAF-protected virtual service on the {partner-product-short-name} instances. To deploy the example application, set the *provisionExampleApp* parameter to *true* and reference the following configuration files via *bigIpRuntimeInitConfig* parameters:

Reviewer: ```suggestion
Optionally, to illustrate how services work, you can deploy an example web application firewall (WAF)–protected web service. The example application deploys web application instances and a WAF-protected virtual service to the {partner-product-short-name} instances. To deploy the example application, set the `provisionExampleApp` parameter to *true*, and reference the following configuration files via `bigIpRuntimeInitConfig` parameters:
```

** `runtime-init-conf-2nic-payg-instance01-with-app.yaml`
** `runtime-init-conf-2nic-payg-instance02-with-app.yaml`

If your deployment requires additional customizations, update and rehost your configuration files at a custom URL.

> NOTE: *IMPORTANT* If hosting on GitHub, URLs that point to GitHub must use the raw file format (for example, `raw.githubusercontent.com`).

Reviewer: ```suggestion
IMPORTANT: If you host on GitHub, URLs that point to GitHub must use the raw-file format (for example, `raw.githubusercontent.com`).
```

> The {partner-product-short-name} configurations must match the external environment in which they are deployed. Although the provided Runtime-Init configuration examples extract and templatize many common values, some values are still hardcoded and must be updated to match your specific environment or deployment first.

Reviewer: ```suggestion
The {partner-product-short-name} configurations must match the external environment in which they deploy. Although the provided `Runtime-Init` configuration examples extract and create a template with many common values, some values are hardcoded and must be manually updated to match your environment.
```

==== Common customization example 1

> To deploy a Bring Your Own License (BYOL) instance, perform these steps:

Reviewer: ```suggestion
To deploy a Bring Your Own License (BYOL) instance, complete the following steps:
```

[arabic]
. Edit or modify the Declarative Onboarding (DO) declaration in a

> corresponding example `byol` runtime-init configuration file with the new `regKey`

Reviewer: Unclear run-on sentence:
_Edit or modify the Declarative Onboarding (DO) declaration in a corresponding example `byol` runtime-init configuration file with the new `regKey` value._

value.

Example snippet:

From:
[source,yaml]
----
...
          My_License:
            class: License
            licenseType: regKey
            regKey: REPLACE_WITH_VALID_REGKEY
...
----

To:
[source,yaml]
----
...
          My_License:
            class: License
            licenseType: regKey
            regKey: AAAAA-BBBBB-CCCCC-DDDDD-EEEEEEE
...
----

[arabic, start=2]
. Publish or host the customized `runtime-init` configuration file at a location

> that can be accessed by {partner-product-short-name} at deployment time (for example, git, S3, etc.).

Reviewer: ```suggestion
that can be accessed by {partner-product-short-name} at deployment time (for example, Git and Amazon S3.).
```

> . Update the *bigIpRuntimeInitConfig* input parameters to reference the

Reviewer: ```suggestion
. Update the `bigIpRuntimeInitConfig` input parameters to reference the
```

new URLs of the updated configurations.

> . Update the *bigIpImageId* input parameter to a valid BYOL image ID.

Reviewer: ```suggestion
. Update the `bigIpImageId` input parameter to a valid BYOL image ID.
```

{empty} +

==== Common customization example 2

> If you want to change host names, perform these steps to disable usage reporting to the {partner-product-short-name} NTP and DNS servers, etc.:

Reviewer: ```suggestion
If you want to change host names, complete the following steps to disable usage reporting to the {partner-product-short-name} NTP and DNS servers:
```

[arabic]
. Edit or modify the DO declaration in a

> corresponding example `runtime-init` configuration file with the new

Reviewer: Unclear. Revise.

values.

Example snippet:

From:
[source,yaml]
----
...
          My_System:
            autoPhonehome: true
            class: System
            hostname: "failover01.local"
...
          failoverGroup:
            class: DeviceGroup
            type: sync-failover
            members:
              - failover01.local
              - failover02.local
...
----

To:
[source,yaml]
----
...
          My_System:
            autoPhonehome: false
            class: System
            hostname: "bigip-cluster-01-a.yourcompany.com"
...
          failoverGroup:
            class: DeviceGroup
            type: sync-failover
            members:
              - bigip-cluster-01-a.yourcompany.com
              - bigip-cluster-01-b.yourcompany.com
...
----

[arabic, start=2]
. Publish or host the customized `runtime-init` configuration files at a location

> that can be accessed by {partner-product-short-name} at deployment time (for example, git, S3, etc.).

Reviewer: ```suggestion
that can be accessed by {partner-product-short-name} at deployment time (for example, GitHub and Amazon S3).
```

> . Update the *bigIpRuntimeInitConfig* input parameters to reference the

Reviewer: ```suggestion
. Update the `bigIpRuntimeInitConfig` input parameters to reference the
```

new URLs of the updated configurations.

{empty} +
{empty} +

> TIP: For additional information and examples, refer to the https://github.com/f5networks/f5-bigip-runtime-init[F5 BIG-IP Runtime Init^] GitHub repository.

Reviewer: ```suggestion
TIP: For more information and examples, refer to the https://github.com/f5networks/f5-bigip-runtime-init[F5 BIG-IP Runtime Init^] GitHub repository.
```
